vhost_user_block/src/lib.rs

```diff
@@ -232,10 +272,38 @@ impl VhostUserBackend for VhostUserBlkBackend {
         match device_event {
             q if device_event < self.config.num_queues => {
                 let mut vring = vrings[q as usize].write().unwrap();
-                if self.process_queue(&mut vring) {
-                    debug!("signalling queue");
-                    vring.signal_used_queue().unwrap();
+
+                if self.poll_queue {
+                    // Actively poll the queue until POLL_QUEUE_US has passed
+                    // without seeing a new request.
+                    let mut now = Instant::now();
+                    loop {
+                        if self.process_queue(&mut vring) {
+                            now = Instant::now();
+                        } else if now.elapsed().as_micros() > POLL_QUEUE_US {
+                            break;
+                        }
+                    }
                 }
+
+                if self.event_idx {
```

Comment:
This branch is already added in process_queue, why check it again here?

Reply:
The check in process_queue is to determine whether the EVENT_IDX feature can be used to omit sending an interrupt to the guest, while the one here, as the comment says, is to tackle the fact that the Queue iterator only checks avail_index once.